Add missing serde/inputFormat combination for Hive

homar opened the pull request:
<!-- Thank you for submitting a pull request! Find more information in our development guide at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md and contact us on #dev in Slack. -->
<!-- Provide an overview of the PR for maintainers and reviewers. -->
## Description
Support for this was accidentally removed in scope of https://github.com/trinodb/trino/pull/18556 (424)



<!-- Provide details that would help an engineer who is unfamiliar with this part of the code. -->
## Additional context and related issues



<!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Add back support for ParquetHiveSerDe with TextInputFormat. ({issue}`issuenumber`)
```

findepi commented:
> This combination was working before so we should keep supporting it.

add information which release or which code change affected it


> ( ) Release notes are required, with the following suggested text:
> 
> ```
> # Section
> * Fix some things. ({issue}`issuenumber`)
> ```

please feel in

electrum reviewed (changes requested):
This works for reads, but what happens if you attempt to write to such a table? I'm worried that this could silently create corrupted data that is not readable. I'd like to understand writing behavior for such a table with

* Current Trino
* Older version of Trino (before the change that prompted this fix)
* Spark

homar commented:
> This works for reads, but what happens if you attempt to write to such a table? I'm worried that this could silently create corrupted data that is not readable. I'd like to understand writing behavior for such a table with
> 
> * Current Trino
> * Older version of Trino (before the change that prompted this fix)
> * Hive
> * Spark

Current trino/previous trino can't write to it as it is a non managed hive table. 
Spark as you can see from tests can read and write.
Hive seems not to able to write to it.

findepi commented:
> Current trino/previous trino can't write to it as it is a non managed hive table.

there is a toggle to enable writes to non-managed tables.

electrum commented:
When you write from Spark, are the files written in Parquet format?

electrum commented:
The test seems to only read with Spark, not write. Can you also read and write with Hive?

homar commented:
@electrum sorry, my bad.
I modified the test.
Now it writes and reads with spark and it validates that the created file is parquet.
Reading and writing with hive doesn't really work, I also added code to validate that.
Writing with trino doesn't work with new version and it didn't work with the old. It fails on this line https://github.com/trinodb/trino/blob/029e1a07ae6823b9ce61eac82ebd7bb65daea966/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java#L3582

electrum reviewed (approved):
Thanks for confirming the behavior. This change looks good. Can you also add a test to verify that writing fails with Trino? I'm assuming something similar to the Hive insert would work.

homar commented:
@electrum can you merge this ? I don't have the button

findepi commented:
> Release notes are required, with the following suggested text:
> 
> ```
> # Section
> * Add back support for ParquetHiveSerDe with TextInputFormat. ({issue}`issuenumber`)
> ```

or maybe "[Hive] Fix reading from certain Parquet tables incorrectly created by Spark"

cc @colebow
